src/components/comments-components/CommentCard.jsx

```diff
@@ -1,12 +1,25 @@
-const CommentCard = ({ comment }) => {
+import { useContext } from "react";
+import DeleteButton from "./DeleteButton";
+import { UserContext } from "../../context/UserContext";
+
+const CommentCard = ({ comment, onDelete }) => {
+
+  const { user } = useContext(UserContext);
+
   return (
-    <div className="p-4 bg-orange-100 border-l-4 border-orange-400 rounded shadow-sm">
+    <div className="p-4 bg-orange-100 border-l-4 border-orange-400 rounded shadow-sm relative">
       <p className="font-semibold text-gray-800">{comment.author}</p>
       <p className="text-gray-700 mt-2">{comment.body}</p>
       <p className="text-gray-800 font-semibold">Votes: {comment.votes}</p>
       <p className="text-sm text-gray-500 mt-1">
         Posted on {new Date(comment.created_at).toLocaleDateString()}
       </p>
+      {user && comment.author === user.username && (
+        <>
+        {console.log('Delete button for comment_id:', comment.comment_id)}
+        <DeleteButton comment_id={comment.comment_id} onDelete={onDelete} />
```

Comment:
You could save having to pass the comment id down by ` onDelete={()=>onDelete(comment.id)} `

Reply:
You may need it for other reasons though, this is just a thought. currently you are passing it down 2 levels of props you may be able to tidy up the abstraction by passing the setComments down one level and then having the delete function declared at the comment level. This is not a prescription by the way more just food for thought.